### PR TITLE
Finish getMarketsInfo

### LINKED ISSUE #1639: Finalize getMarketsInfo & add more tests
`getMarketsInfo` requires some log changes before it can be finished. Also need to add more tests to check these new values

### PR DESCRIPTION
**Pull request by adrake33**
Fixes https://github.com/AugurProject/augur/issues/1639. Adds code and new checks for a number of values returned by `getMarketsInfo`, including `outcomePrices`, `reportingState`, `consensus`, `needsMigration`, `finalizationBlockNumber`, & `finalizationTime`

### CODE REVIEW

File: packages/augur-state/src/db/DB.ts [hunk 1]
```diff
@@ -348,6 +351,28 @@ export class DB<TBigNumber> {
...
 
+    /**
+   * Queries the DisputeCrowdsourcerContribution DB
+   *
+   * @param {PouchDB.Find.FindRequest<{}>} request Query object
+   * @returns {Promise<Array<DisputeCrowdsourcerContributionLog>>}
+   */
+  public async findDisputeCrowdsourcerContributionLogs(request: PouchDB.Find.FindRequest<{}>): Promise<Array<DisputeCrowdsourcerContributionLog>> {
+    const results = await this.findInSyncableDB(this.getDatabaseName("DisputeCrowdsourcerContribution"), request);
+    return results.docs as unknown as Array<DisputeCrowdsourcerContributionLog>;
```
Comment (brianosaurus) [inline packages/augur-state/src/db/DB.ts:362]: How'd you figure out to do as twice? Whoa...never thought of that.

File: packages/augur-test/libs/ContractAPI.ts [hunk 2]
```diff
@@ -178,11 +185,18 @@ export class ContractAPI {
     await market.contribute(payoutNumerators, amount, "");
   }
 
-  public derivePayoutDistributionHash(market: GenericAugurInterfaces.Market<ethers.utils.BigNumber>, payoutNumerators: Array<ethers.utils.BigNumber>): Promise<string> {
+  public async getRemainingToFill(market: GenericAugurInterfaces.Market<ethers.utils.BigNumber>, payoutNumerators: Array<ethers.utils.BigNumber>): Promise<ethers.utils.BigNumber> {
+    const payoutDistributionHash = await this.derivePayoutDistributionHash(market, payoutNumerators);
+    const crowdsourcerAddress = await market.getCrowdsourcer_(payoutDistributionHash);
+    const crowdsourcer = this.augur.contracts.getReportingParticipant(crowdsourcerAddress);
+    return await crowdsourcer.getRemainingToFill_();
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: I found this regarding return await: https://eslint.org/docs/rules/no-return-await

File: packages/augur-test/libs/ContractAPI.ts [hunk 3]
```diff
@@ -85,7 +90,9 @@ export class ContractAPI {
   }
 
   public async createReasonableScalarMarket(universe: GenericAugurInterfaces.Universe<ethers.utils.BigNumber>, outcomes: Array<string>): Promise<GenericAugurInterfaces.Market<ethers.utils.BigNumber>> {
-    const endTime = new ethers.utils.BigNumber(Math.round(new Date().getTime() / 1000) + 30 * 24 * 60 * 60);
+    const time = this.augur.contracts.getTime();
+    let currentTimestamp = (await time.getTimestamp_()).toNumber();
+    const endTime = new ethers.utils.BigNumber(currentTimestamp + 30 * 24 * 60 * 60);
     const fee = new ethers.utils.BigNumber(10).pow(new ethers.utils.BigNumber(16));
     const affiliateFeeDivisor = new ethers.utils.BigNumber(25);
     return await this.createScalarMarket(universe, outcomes, endTime, fee, affiliateFeeDivisor, this.account);
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: https://eslint.org/docs/rules/no-return-await

File: packages/augur-test/libs/ContractAPI.ts [hunk 4]
```diff
@@ -63,7 +66,9 @@ export class ContractAPI {
   }
 
   public async createReasonableMarket(universe: GenericAugurInterfaces.Universe<ethers.utils.BigNumber>, outcomes: Array<string>): Promise<GenericAugurInterfaces.Market<ethers.utils.BigNumber>> {
-    const endTime = new ethers.utils.BigNumber(Math.round(new Date().getTime() / 1000) + 30 * 24 * 60 * 60);
+    const time = this.augur.contracts.getTime();
+    let currentTimestamp = (await time.getTimestamp_()).toNumber();
+    const endTime = new ethers.utils.BigNumber(currentTimestamp + 30 * 24 * 60 * 60);
     const fee = new ethers.utils.BigNumber(10).pow(new ethers.utils.BigNumber(16));
     const affiliateFeeDivisor = new ethers.utils.BigNumber(25);
     return await this.createCategoricalMarket(universe, outcomes, endTime, fee, affiliateFeeDivisor, this.account);
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: same as below (sorry doing this in reverse)

File: packages/augur-test/libs/ContractAPI.ts [hunk 5]
```diff
@@ -44,7 +45,9 @@ export class ContractAPI {
   }
 
   public async createReasonableYesNoMarket(universe: GenericAugurInterfaces.Universe<ethers.utils.BigNumber>): Promise<GenericAugurInterfaces.Market<ethers.utils.BigNumber>> {
-    const endTime = new ethers.utils.BigNumber(Math.round(new Date().getTime() / 1000) + 30 * 24 * 60 * 60);
+    const time = this.augur.contracts.getTime();
+    let currentTimestamp = (await time.getTimestamp_()).toNumber();
+    const endTime = new ethers.utils.BigNumber(currentTimestamp + 30 * 24 * 60 * 60);
     const fee = new ethers.utils.BigNumber(10).pow(new ethers.utils.BigNumber(16));
     const affiliateFeeDivisor = new ethers.utils.BigNumber(25);
     return await this.createYesNoMarket(universe, endTime, fee, affiliateFeeDivisor, this.account);
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: see below

File: packages/augur-test/libs/ContractAPI.ts [hunk 6]
```diff
@@ -201,6 +215,12 @@ export class ContractAPI {
     return this.augur.contracts.disputeWindowFromAddress(disputeWindowAddress);
   }
 
+  public async getDisputeWindowEndTime(market: GenericAugurInterfaces.Market<ethers.utils.BigNumber>): Promise<ethers.utils.BigNumber> {
+    const disputeWindowAddress = await market.getDisputeWindow_();
+    const disputeWindow = this.augur.contracts.disputeWindowFromAddress(disputeWindowAddress);
+    return await disputeWindow.getEndTime_();
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: same

File: packages/augur-test/libs/ContractAPI.ts [hunk 7]
```diff
@@ -229,6 +249,13 @@ export class ContractAPI {
     await market.doInitialReport(payoutNumerators, "");
   }
 
+  public async getInitialReporterStake(market: GenericAugurInterfaces.Market<ethers.utils.BigNumber>, payoutNumerators: Array<ethers.utils.BigNumber>): Promise<ethers.utils.BigNumber> {
+    const payoutDistributionHash = await this.derivePayoutDistributionHash(market, payoutNumerators);
+    const initialReporterAddress = await market.getCrowdsourcer_(payoutDistributionHash);
+    const initialReporter = this.augur.contracts.getReportingParticipant(initialReporterAddress);
+    return await initialReporter.getStake_();
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts]: same

File: packages/augur-test/libs/ContractAPI.ts [hunk 8]
```diff
@@ -229,6 +249,13 @@ export class ContractAPI {
...
+  }
+
   public async finalizeMarket(market: GenericAugurInterfaces.Market<ethers.utils.BigNumber>): Promise<void> {
     await market.finalize();
```
Comment (brianosaurus) [inline packages/augur-test/libs/ContractAPI.ts:259]: Unsure if that does anything?

**Review by brianosaurus — APPROVED** [review]
Found some stuff that maybe you might want to change? Technically correct but patterns we leave behind tend to get copied by other as you know. I'm not sure if some of the code is necessary. You know typescript/js better than I do however.

File: packages/augur-state/src/db/DB.ts (hunk 1 above)
Comment (adrake33) [reply packages/augur-state/src/db/DB.ts:362]: Haha, I think Alex was the one who originally wrote this. I copied this from his original function to create these new find functions.

File: packages/augur-test/libs/ContractAPI.ts (hunk 2 above)
Comment (adrake33) [reply packages/augur-test/libs/ContractAPI.ts]: Oh, interesting! We're actually doing this in a fair number of places throughout the code, but I'll update these.

File: packages/augur-test/libs/ContractAPI.ts (hunk 8 above)
Comment (adrake33) [reply packages/augur-test/libs/ContractAPI.ts:259]: This calls the `finalize` function on the Market.sol contract. There's no return value that it needs to return, though.

**Pull request merged**